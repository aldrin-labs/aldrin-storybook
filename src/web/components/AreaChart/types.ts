export interface Props {
  data: {
    x: number
    y: number
  }[]
}
