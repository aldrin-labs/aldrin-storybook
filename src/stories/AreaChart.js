import React from 'react'

import { storiesOf } from '@storybook/react'
import { withInfo } from '@storybook/addon-info'

import AreaChart from '@components/AreaChart'

const data = [{ x: 1, y: 2 }, { x: 2, y: 4 }]

storiesOf('AreaChart', module)
  .add(
    'AreaChart',
    withInfo({ inline: true })(() =>
      <AreaChart data={data} />
    )
  )
