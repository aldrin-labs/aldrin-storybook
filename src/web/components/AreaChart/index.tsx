import React, { Component } from 'react'
import styled from 'styled-components'
import { XAxis, YAxis, AreaSeries, FlexibleXYPlot } from 'react-vis'

import { Props } from './types'

export class AreaChart extends Component<Props> {
  render() {
    const { data } = this.props

    return (
      <Chart>
        <FlexibleXYPlot>
          <XAxis hideLine hideTicks />
          <YAxis hideLine hideTicks />

          <AreaSeries
            data={data}
            style={{
              fill: 'rgba(133, 237, 238, 0.35)',
              stroke: 'rgb(78, 216, 218)',
              strokeWidth: '3px',
            }}
          />
        </FlexibleXYPlot>
      </Chart>
    )
  }
}

export default AreaChart

const Chart = styled.div`
  width: 100%;
  height: 5em;
  min-height: 5em;
  max-height: 10em;
  margin-top: 24px;
`
