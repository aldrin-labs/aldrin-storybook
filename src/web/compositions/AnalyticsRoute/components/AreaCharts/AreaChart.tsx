import React, { useEffect } from 'react'
import { WhiteTitle, HeaderContainer, Row, ChartContainer } from '../../index.styles'
import { createAreaChart } from '../utils'
import { Theme } from '@material-ui/core'
// import { TooltipForAreaChart } from './Tooltips'

const AreaChart = ({ theme, data, selectedPair, isDataLoading, id }: { theme: Theme, data: any, selectedPair: string }) => {
  useEffect(() => {
    createAreaChart(data)

    return () => window.myAreaChart.destroy()
  }, [id])

  return (
    <>
      <HeaderContainer theme={theme} justify={'space-between'}>
        <Row margin={'0 0 0 2rem'}>
          <WhiteTitle theme={theme}>Volume</WhiteTitle>
        </Row>
        <Row margin={'0 2rem 0 0'}>
          <WhiteTitle theme={theme}>30d</WhiteTitle>
        </Row>
      </HeaderContainer>
      <ChartContainer>
        <canvas id="areaChart"></canvas>
        {/* <TooltipForAreaChart theme={theme} /> */}
      </ChartContainer>
    </>
  )
}

export default AreaChart
