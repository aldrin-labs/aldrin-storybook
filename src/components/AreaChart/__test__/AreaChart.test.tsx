import * as React from 'react'
import { shallow, mount } from 'enzyme'
import AreaChart from '../index'

describe('Switch', () => {
  it('render()', () => {
    const wrapper = shallow(
      <AreaChart data={[{ x: 1, y: 2 }, { x: 2, y: 3 }]} />
    )

    expect(wrapper).toMatchSnapshot()
  })
})
