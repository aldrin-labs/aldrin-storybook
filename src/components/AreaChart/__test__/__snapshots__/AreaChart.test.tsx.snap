// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch render() 1`] = `
<[object Object]>
  <FlexibleXYPlot>
    <XAxis
      attr="x"
      attrAxis="y"
      hideLine={true}
      hideTicks={true}
      orientation="bottom"
    />
    <YAxis
      attr="y"
      attrAxis="x"
      hideLine={true}
      hideTicks={true}
      orientation="left"
    />
    <AreaSeries
      className=""
      data={
        Array [
          Object {
            "x": 1,
            "y": 2,
          },
          Object {
            "x": 2,
            "y": 3,
          },
        ]
      }
      getNull={[Function]}
      stack={false}
      style={
        Object {
          "fill": "rgba(133, 237, 238, 0.35)",
          "stroke": "rgb(78, 216, 218)",
          "strokeWidth": "3px",
        }
      }
    />
  </FlexibleXYPlot>
</[object Object]>
`;
